src/App.tsx

```diff
@@ -1,26 +1,272 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { Errors } from './utils/Errors/Errors';
+import { Footer } from './utils/Footer/Footer';
+import { Header } from './utils/Header/Header';
+import { TodoList } from './utils/TodoList/TodoList';
+import { deleteTodos, getTodos, patchTodos, postTodos } from './api/todos';
 
-const USER_ID = 0;
+const USER_ID = 700;
+
+enum Filter {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string>('');
+  const [filter, setFilter] = useState<string>(Filter.All);
+  const [newTodoTitle, setNewTodoTitle] = useState<string>('');
+  const [allCompleted, setAllCompleted] = useState<boolean>(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  // const [loadingTodoId, setLoadingTodoId] = useState<number | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const completedTodos = todos.filter(todo => todo.completed).length;
+  const notCompletedTodos = todos.filter(todo => !todo.completed).length;
+  const areAllCompleted =
+    todos?.length > 0 && todos?.every(todo => todo.completed);
+
+  // #region get
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setError('Unable to load todos'));
+  }, []);
+
+  const getVisibleTodos = (generalTodos: Todo[], generalFilter: string) => {
+    let filteredTodos = generalTodos;
+
+    if (generalFilter === Filter.Active) {
+      filteredTodos = filteredTodos.filter(todo => !todo.completed);
+    } else if (generalFilter === Filter.Completed) {
+      filteredTodos = filteredTodos.filter(todo => todo.completed);
+    }
+
+    return filteredTodos;
+  };
+
+  const visibleTodos: Todo[] = useMemo(() => {
+    return getVisibleTodos(todos, filter);
+  }, [todos, filter]);
+
+  // #endregion
+
+  //#region patchStatus
+  const handleToggleAllCompleted = () => {
+    const arrayWithNotCompletedTodoIds = visibleTodos
+      .filter(todo => !todo.completed)
+      .map(todo => todo.id);
+
+    const arrayWithCompletedTodoIds = visibleTodos
+      .filter(todo => todo.completed)
+      .map(todo => todo.id);
+
+    const updatedTodos = todos.map(todo => ({
+      ...todo,
+      completed: !allCompleted,
+    }));
+
+    if (allCompleted) {
+      setLoadingTodoIds(arrayWithCompletedTodoIds);
+    } else {
+      setLoadingTodoIds(arrayWithNotCompletedTodoIds);
+    }
+
+    Promise.all(
```

Comment:
Generally using Promise.all in such cases might be ok
But to pass tests just iterate through all todo ids you need to delete (forEach) and send requests separately. In this case it'll allow to handle each erroneous result separately